solidity/contracts/token/libs/TokenRouter.sol

```diff
@@ -175,17 +176,28 @@ abstract contract TokenRouter is GasRouter {
     /**
      * @notice Returns the gas payment required to dispatch a message to the given domain's router.
      * @param _destinationDomain The domain of the router.
-     * @return _gasPayment Payment computed by the registered InterchainGasPaymaster.
+     * @return quotes Payment computed by the registered InterchainGasPaymaster.
      */
-    function quoteGasPayment(
-        uint32 _destinationDomain
-    ) public view override returns (uint256) {
-        return
-            _GasRouter_quoteDispatch(
-                _destinationDomain,
-                TokenMessage.format(bytes32(0), type(uint256).max, bytes("")),
-                address(hook)
-            );
+    function quoteTransferRemote(
```

Comment:
Is this basically for non-collateral tokenrouters? if so, probably useful to indicate in a comment